riscv-c-api.md

```diff
@@ -116,5 +116,41 @@ For example:
 
 ## Intrinsic Functions
 
-Do we really have none of these?  I can't figure out
-`gcc/gcc/config/riscv/riscv-builtins.c`...
+RISC-V intrinsics allow access to RISC-V-specific instructions from C or C++ that are usually not otherwise synthesizable by compilers.
+Intrinsics typically map to a single instruction that is inlined to avoid function call overhead.
+
+Intrinsics are only available if the corresponding instructions are available (as part of the effective `-march` string).
+Some intrinsics are only available if a particular header file is included.
+RISC-V header files, that enable intrinsics, require the prefix `riscv_` (e.g. `riscv_vector.h` or `riscv_crypto.h`).
+
+RISC-V intrinsics follow the following naming rule:
+
+```
+INTRINSIC ::= PREFIX '_' MNEMONIC [ '_' RET_TYPE ]
+PREFIX ::= "__builtin_riscv" | "__rv" EXTENSION | "_rv" | "_rv32" | "_rv64"
+EXTENSION ::= ISA extension letter or string.
+MNEMONIC ::= Instruction name in the ISA extensions specification. Replace '.' with '_' (if any).
+RET_TYPE ::= Return type indication.
+```
```

Comment:
In vector intrinsic, there is some naming [exception ](https://github.com/riscv-non-isa/rvv-intrinsic-doc/blob/master/rvv-intrinsic-rfc.md#exceptions-in-naming) cases due to having the same return type under different input types. 
I was thinking maybe naming rule need allow the exception or only require  `PREFIX` and `MNEMONIC` part should be common?